### DISCUSSION TITLE
SRFI-42 and its use of ":"

### DISCUSSION THREAD

**Discussion by jpellegrini** [egallesio/STklos#206] [Ideas]
Hi @egallesio !

SRFI 42 uses identifiers beginning with ":", which STkos treats as keywords, and not symbols.

What if we implement SRFI-42 with the following changes to STklos:

- The identifiers in SRFI-42 may be used as variables (only those, and only after SRFI-42 is loaded);
- No other changes to STklos -- `:x` would still be a keyword, and not allowed as variable name.

The identifiers are:

```
:         ;;  the single ":" identifier!
:list
:string
:vector
:range
:real-range
:char-range
:port
:integers
:dispatched
:do
:let
:parallel
:while
:until
```

Do you think it's worth it?

**Comment by lassik** [comment]
The custom around SRFIs is that implementers are allowed to add extra features as long as they don't conflict with anything specified in the SRFI.

SRFI 42 doesn't say anything about keywords. If I'm not mistaken, this means we could implement it so that the keywords `:list :string :vector ...` are equivalent to the symbols with those names. So when someone uses SRFI 42 on STklos, they can write either `:list` or `|:list|` and the result will be the same.
